httpmock.go

```diff
@@ -81,6 +80,13 @@ type HandlerWithHeaders interface {
 	HandleWithHeaders(method, path string, headers http.Header, body []byte) Response
 }
 
+// NewHandler returns a pointer to a new handler with the test struct set
+func NewHandler(t *testing.T) *MockHandler {
```

Comment:
Should this be called NewMockHandler? We probably also want to do the following:
- Add NewMockHandlerWithHeaders
- Update the examples/docs to use New instead of instantiating it as it does now